patches/minecraft/net/minecraft/entity/EntityLivingBase.java.patch

```diff
@@ -231,15 +231,17 @@
          super.func_70071_h_();
          this.func_184608_ct();
  
-@@ -2015,6 +2046,7 @@
+@@ -2014,7 +2045,9 @@
+ 
                  if (!ItemStack.func_77989_b(itemstack1, itemstack))
                  {
++                    if (!ItemStack.areItemStacksEqualUsingNBTShareTag(itemstack1, itemstack))
```

Comment:
Yeah I think you're right, not all "real" changes will get synced to the client.